rfc/src/rfcs/0002-function-stubbing.md

```diff
@@ -161,16 +161,17 @@ Given a set of `original`-`replacement` pairs, Kani will exit with an error if
 ### Stub compatibility and validation
 
 When considering whether a function/method can be replaced with some given stub, we want to allow some measure of flexibility, while also ensuring that we can provide the user with useful feedback if stubbing results in misformed code.
-We consider a stub to be compatible with some function/method if all the following conditions are met:
+We consider a stub and a function/method to be compatible if all the following conditions are met:
 
-- They have the same return type. The names of type variables need to match.
 - They have the same number of parameters.
+- They have the same return type.
 - Each parameter in the stub has the same type as the corresponding parameter in the original function/method.
-The names of type variables need to match.
-- During monomorphization, every statically dispatched call to a trait method can be resolved.
+- We do **NOT** allow for the renaming of type parameters when checking whether types are equivalent; for example, the function `foo<S>(x: S) -> S` is considered to have a different type than the function `bar<T>(x: T) -> T`.
```

Comment:
Oh, why is that?

Reply:
Not for a fundamental reason, and I've updated the RFC to be more flexible here: the stub and original function/method need to have the same number of generic parameters, but they can be named different things.